homeassistant/components/myuplink/coordinator.py

```diff
@@ -0,0 +1,65 @@
+"""Coordinator for myUplink."""
+import asyncio.timeouts
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+
+from myuplink.api import MyUplinkAPI
+from myuplink.models import Device, DevicePoint, System
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class CoordinatorData:
+    """Represent coordinator data."""
+
+    systems: list[System]
+    devices: dict[str, Device]
+    points: dict[str, dict[str, DevicePoint]]
+    time: datetime
+
+
+class MyUplinkDataCoordinator(DataUpdateCoordinator[CoordinatorData]):
+    """Coordinator for myUplink data."""
+
+    def __init__(self, hass: HomeAssistant, api: MyUplinkAPI) -> None:
+        """Initialize myUplink coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="myuplink",
+            update_interval=timedelta(seconds=60),
+        )
+        self.api = api
+
+    async def _async_update_data(self) -> CoordinatorData:
+        """Fetch data from the myUplink API."""
+        async with asyncio.timeout(10):
+            # Get systems
+            systems = await self.api.async_get_systems()
+
+            devices: dict[str, Device] = {}
+            points: dict[str, dict[str, DevicePoint]] = {}
+            device_ids = [
```

Comment:
Are the device ids unique even when coming from different systems?

Reply:
I can't find something that confirms it, but when i'm looking at the values it seems to be unique values/ids.